bundles/org.openhab.binding.pushbullet/src/main/java/org/openhab/binding/pushbullet/internal/handler/PushbulletHandler.java

```diff
@@ -0,0 +1,229 @@
+
+    private final Logger logger = LoggerFactory.getLogger(PushbulletHandler.class);
+
+    private final Gson gson = new GsonBuilder().create();
+
+    private static final Version VERSION = FrameworkUtil.getBundle(PushbulletHandler.class).getVersion();
+
+    private static final Pattern CHANNEL_PATTERN = Pattern.compile("^[a-zA-Z0-9_-]+$");
+
+    private @Nullable PushbulletConfiguration config;
+
+    public PushbulletHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("About to handle {} on {}", command, channelUID);
+
+        // Future improvement: If recipient is already set, send a push on a command channel change
+        // check reconnect channel of the unifi binding for that
+
+        logger.debug("The Pushbullet binding is a read-only binding and cannot handle command '{}'.", command);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+        config = getConfigAs(PushbulletConfiguration.class);
+
+        // Name and Token are both "required", so set the Thing immediately ONLINE.
+        updateStatus(ThingStatus.ONLINE);
+
+        logger.debug("Finished initializing!");
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(PushbulletActions.class);
+    }
+
+    public boolean sendPush(@Nullable String recipient, @Nullable String message, String type) {
+        return sendPush(recipient, "", message, type);
+    }
+
+    public boolean sendPush(@Nullable String recipient, @Nullable String title, @Nullable String message, String type) {
+        boolean result = false;
+
+        logger.debug("sendPush is called for ");
+        logger.debug("Thing {}", thing);
+        logger.debug("Thing Label: '{}'", thing.getLabel());
+
+        PushbulletConfiguration configuration = getConfigAs(PushbulletConfiguration.class);
+        logger.debug("CFG {}", configuration);
+
+        Properties headers = prepareRequestHeaders(configuration);
+
+        String request = prepareMessageBody(recipient, title, message, type);
+
+        try (InputStream stream = new ByteArrayInputStream(request.getBytes(StandardCharsets.UTF_8))) {
+
+            String pushAPI = configuration.getApiUrlBase() + "/" + API_METHOD_PUSHES;
+
+            String responseString = HttpUtil.executeUrl(HttpMethod.POST.asString(), pushAPI, headers, stream,
+                    MimeTypes.Type.APPLICATION_JSON.asString(), TIMEOUT);
+
+            logger.debug("Got Response: {}", responseString);
+            PushResponse response = gson.fromJson(responseString, PushResponse.class);
+
+            logger.debug("Unpacked Response: {}", response);
+
+            stream.close();
+
+            if ((null != response) && (null == response.getPushError())) {
+                result = true;
+            }
+        }
```

Comment:
Formatting is a little bit a-typical

Reply:
Maybe caused by my IntelliJ installation. It should have applied the Eclipse formatting rules though.

Can I fix that in an upcoming PR when I touch this code again?